shell/platform/embedder/tests/embedder_test_gl_surface.cc

```diff
@@ -0,0 +1,228 @@
+                                  config,             // config
+                                  surface_attributes  // surface attributes
+        );
+    FML_CHECK(offscreen_surface_ != EGL_NO_SURFACE) << GetEGLError();
+  }
+
+  {
+    const EGLint context_attributes[] = {
+        EGL_CONTEXT_CLIENT_VERSION,  //
+        2,                           //
+        EGL_NONE                     //
+    };
+
+    onscreen_context_ =
+        ::eglCreateContext(display_,           // display connection
+                           config,             // config
+                           EGL_NO_CONTEXT,     // sharegroup
+                           context_attributes  // context attributes
+        );
+    FML_CHECK(onscreen_context_ != EGL_NO_CONTEXT) << GetEGLError();
+
+    offscreen_context_ =
+        ::eglCreateContext(display_,           // display connection
+                           config,             // config
+                           onscreen_context_,  // sharegroup
+                           context_attributes  // context attributes
+        );
+    FML_CHECK(offscreen_context_ != EGL_NO_CONTEXT) << GetEGLError();
+  }
+}
+
+EmbedderTestGLSurface::~EmbedderTestGLSurface() {
+  auto result = ::eglDestroyContext(display_, onscreen_context_);
+  FML_CHECK(result == EGL_TRUE) << GetEGLError();
+
+  result = ::eglDestroyContext(display_, offscreen_context_);
+  FML_CHECK(result == EGL_TRUE) << GetEGLError();
+
+  result = ::eglDestroySurface(display_, onscreen_surface_);
+  FML_CHECK(result == EGL_TRUE) << GetEGLError();
+
+  result = ::eglDestroySurface(display_, offscreen_surface_);
+  FML_CHECK(result == EGL_TRUE) << GetEGLError();
+
+  result = ::eglTerminate(display_);
+  FML_CHECK(result == EGL_TRUE);
+}
+
+bool EmbedderTestGLSurface::MakeCurrent() {
+  auto result = ::eglMakeCurrent(display_, onscreen_surface_, onscreen_surface_,
+                                 onscreen_context_);
+
+  if (result == EGL_FALSE) {
+    FML_LOG(ERROR) << "Could not make the context current. " << GetEGLError();
+  }
+
+  return result == EGL_TRUE;
+}
+
+bool EmbedderTestGLSurface::ClearCurrent() {
+  auto result = ::eglMakeCurrent(display_, EGL_NO_SURFACE, EGL_NO_SURFACE,
+                                 EGL_NO_CONTEXT);
+
+  if (result == EGL_FALSE) {
+    FML_LOG(ERROR) << "Could not clear the current context. " << GetEGLError();
+  }
+
+  return result == EGL_TRUE;
+}
+
+bool EmbedderTestGLSurface::Present() {
+  auto result = ::eglSwapBuffers(display_, onscreen_surface_);
+
+  if (result == EGL_FALSE) {
+    FML_LOG(ERROR) << "Could not swap buffers. " << GetEGLError();
+  }
+
+  return result == EGL_TRUE;
+}
+
+uint32_t EmbedderTestGLSurface::GetFramebuffer() {
+  // Return FBO0
+  return 0;
+}
+
+bool EmbedderTestGLSurface::MakeResourceCurrent() {
+  auto result = ::eglMakeCurrent(display_, offscreen_surface_,
+                                 offscreen_surface_, offscreen_context_);
+
+  if (result == EGL_FALSE) {
```

Comment:
There are many
```
  if (result == EGL_FALSE) {
    FML_LOG(ERROR) << "Could not XXX. " << GetEGLError();
  }

   return result == EGL_TRUE;
```

Maybe we can shorten it by `return CheckResult(result, "XXX");` with
```
bool CheckResult(result, action_name) {
  if (result == EGL_FALSE) {
    FML_LOG(ERROR) << "Could not " << action_name << ". " << GetEGLError();
  }
   return result == EGL_TRUE;
}
```

Reply:
This should be extent of the use of this pattern. So I'd rather keep this as is for now. If we need to add more such uses, I'll patch the same to use the recommended patterh.